src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
```

Comment:
@jamen The only answer would be: test it 🤓 I can only be sure for what I measure: `v8`.

But I will make the assumption (and I don't really like it) that as it's a cross language pattern, known way before JS became something serious, and as compilers exist for a long time and address those kind of issues since I had my first teeth, I'm pretty confident that any first-grade compiler will do it. But again, only tests can speak.

Also as pointed out in the article, accessing an array triggers a bound check, that itself requires a `length` lookup. I'm pretty the algorithm is detailed in the specs. If it's in the specs, every engine will do it. And if every engine lookup for `length` when accessing an array, end of story.